src/core/expression/qgsexpressioncontextutils.cpp

```diff
@@ -535,6 +535,13 @@ QgsExpressionContextScope *QgsExpressionContextUtils::mapToolCaptureScope( const
   return scope;
 }
 
+QgsExpressionContextScope *QgsExpressionContextUtils::mapLayerPositionScope( const QgsPointXY &position )
+{
+  QgsExpressionContextScope *scope = new QgsExpressionContextScope( QObject::tr( "Map Layer Position" ) );
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "layer_position" ), QVariant::fromValue( QgsGeometry::fromPointXY( position ) ) ) );
```

Comment:
We have an existing variable "@canvas_cursor_point". Can we make this "@layer_cursor_point" to follow that same naming style?

This also needs to be added to QgsExpression::initVariableHelp()

Reply:
No problem.